src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1307,16 +1310,19 @@ public virtual bool CanApplyParseOptionChange(ParseOptions oldOptions, ParseOpti
         /// </summary>
         protected virtual void ApplyProjectChanges(ProjectChanges projectChanges)
         {
+            // It's OK to use the null-suppression operator when calling ApplyCompilation/ParseOptionsChanged: the only change that is allowed
+            // is going from one non-null value to another which is blocked by the Project.WithCompilationOptions() API directly.
```

Comment:
I don't understand this comment. Is from one non-null to another allowed or not?

Reply:
Yeah, misphrased. You can't use WithCompilationOptions to make something that was non-null null, hence we know here either both sides are null or both sides are non-null.